### PR TITLE
reduce conversion and gtests

### PR DESCRIPTION
**Pull request by amystamile-usgs**
reduce conversion and gtests

## Description
converted main.cpp to reduce_app.cpp and reduce_app.h. Added 6 gtests. 

## Related Issue
Test conversion

## Motivation and Context
61 MBs

## How Has This Been Tested?
6 gtests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/apps/reduce/reduce_app.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#ifndef reduceapp_h // Change this to your app name in all lower case suffixed with _h (e.g. campt_h, cam2map_h etc.)
```
Comment (kaitlyndlee) [inline isis/src/base/apps/reduce/reduce_app.h]: Left over comment from copy/paste?

File: isis/tests/FunctionalTestsReduce.cpp [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  QVector<QString> args = {"from=" + testCube->fileName(),
+                            "to=" + outCubeFileName,
+                            "algorithm=average",
+                            "mode=total",
+                            "ons=100",
+                            "onl=100"
+                          };
+
+  UserInterface options(APP_XML, args);
+  Pvl log;
```
Comment (kaitlyndlee) [inline isis/tests/FunctionalTestsReduce.cpp]: Since we don't use the log in the tests, can we remove the log variables?

Comment (amystamile-usgs) [reply isis/tests/FunctionalTestsReduce.cpp]: I believe the reason we have log variables in the tests is due to needing to run reduce_app which requires a log.

Comment (kaitlyndlee) [reply isis/tests/FunctionalTestsReduce.cpp]: Because log defaults to a nullptr, if you don't pass one in, it should set to null.

Comment (amystamile-usgs) [reply isis/tests/FunctionalTestsReduce.cpp]: An alternative would be to specify a default value of nullptr in the header and remove the log variables in the tests. Let me know if you had something else in mind.

Comment (kaitlyndlee) [reply isis/tests/FunctionalTestsReduce.cpp]: +1 That's exactly what I was thinking.

**Review by kaitlyndlee — APPROVED** [review]
Everything looks good!

**Pull request merged**